### PR TITLE
Clarify checkpoint deprecation message

### PR DESCRIPTION
**Pull request by edenlightning**
Make the deprecation message of checkpoint_callback arg clearer. Follow up to https://github.com/PyTorchLightning/pytorch-lightning/pull/4461/

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 1]
```diff
@@ -59,7 +59,8 @@ def configure_checkpoint_callbacks(self, checkpoint_callback: Union[ModelCheckpo
             # TODO: deprecated, remove this block in v1.3.0
             rank_zero_warn(
                 "Passing a ModelCheckpoint instance to Trainer(checkpoint_callbacks=...)"
-                " is deprecated since v1.1 and will no longer be supported in v1.3.",
+                " is deprecated since v1.1 and will no longer be supported in v1.3."
+                "Use `callbacks` argument instead.",
```
Comment (awaelchli) [inline pytorch_lightning/trainer/connectors/callback_connector.py:63]:
```suggestion
                " Use `callbacks` argument instead.",
```
needs the space after period.

**Review by awaelchli — APPROVED** [review]
thanks @edenlightning

File: docs/source/trainer.rst [hunk 2]
```diff
@@ -580,9 +580,17 @@ You can override the default behavior by initializing the :class:`~pytorch_light
 callback, and adding it to the :paramref:`~pytorch_lightning.trainer.trainer.Trainer.callbacks` list.
 See :ref:`Saving and Loading Weights <weights_loading>` for how to customize checkpointing.
 
+.. code-block:: python
```
Comment (Borda) [inline docs/source/trainer.rst:583]:
```suggestion
.. testcode::
```

**Review by Borda — COMMENTED** [review]
so it shall be 1.1 milestone, right?

File: docs/source/trainer.rst (hunk 2 above)
Comment (edenlightning) [reply docs/source/trainer.rst:583]: why?

**Comment by edenlightning** [comment]
> so it shall be 1.1 milestone, right?

I don't think so. This is not a feature. Why shouldn't it be in 1.0.7?

**Comment by Borda** [comment]
> > so it shall be 1.1 milestone, right?
> 
> I don't think so. This is not a feature. Why shouldn't it be in 1.0.7?

then the msg does not fit as you say deprecate in 1.1 but you want to release it before 1.1... in such case, we shall rephrase it as deprecated in 1.0.7 and removed in 1.2

File: docs/source/trainer.rst (hunk 2 above)
Comment (Borda) [reply docs/source/trainer.rst:583]: this should try to execute the test, which will be testing that the docs are aligned with codebase :]

**Review by ydcjeff — APPROVED** [review]

**Comment by edenlightning** [comment]
Well this is already in 1.0.5? https://github.com/PyTorchLightning/pytorch-lightning/pull/4336
(Whats the fastest way to verify?)

If so I guess it should be 1.0.5. Altough maybe not super elegant to add deprecations is patch releases :0

**Review by s-rog — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

File: docs/source/trainer.rst [hunk 3]
```diff
@@ -580,9 +580,17 @@ You can override the default behavior by initializing the :class:`~pytorch_light
 callback, and adding it to the :paramref:`~pytorch_lightning.trainer.trainer.Trainer.callbacks` list.
 See :ref:`Saving and Loading Weights <weights_loading>` for how to customize checkpointing.
 
+.. testcode::
+
+    # Init ModelCheckpoint callback, monitoring 'val_loss'
```
Comment (Borda) [inline docs/source/trainer.rst:585]:
```suggestion
    from pytorch_lightning.callbacks import ModelCheckpoint
    # Init ModelCheckpoint callback, monitoring 'val_loss'
```

File: docs/source/trainer.rst [hunk 4]
```diff
@@ -580,9 +580,18 @@ You can override the default behavior by initializing the :class:`~pytorch_light
...
+    checkpoint_callback = ModelCheckpoint(monitor='val_loss')
+
+    # Add your callback to the callbacks list
+    trainer = Trainer(callbacks=[checkpoint_callback])
+
 
 .. warning:: Passing a ModelCheckpoint instance to this argument is deprecated since
-    v1.1.0 and will be unsupported from v1.3.0.
+    v1.1.0 and will be unsupported from v1.3.0. Use `callbacks` argument instead.
```
Comment (Borda) [inline docs/source/trainer.rst:594]:
```suggestion
    v1.1 and will be unsupported from v1.3. Use `callbacks` argument instead.
```

File: pytorch_lightning/trainer/trainer.py [hunk 5]
```diff
@@ -179,7 +179,7 @@ def __init__(
                 :paramref:`~pytorch_lightning.trainer.trainer.Trainer.callbacks`. Default: ``True``.
 
                 .. warning:: Passing a ModelCheckpoint instance to this argument is deprecated since
-                    v1.1.0 and will be unsupported from v1.3.0.
+                    v1.1.0 and will be unsupported from v1.3.0. Use `callbacks` argument instead.
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py:182]:
```suggestion
                    v1.1 and will be unsupported from v1.3. Use `callbacks` argument instead.
```

**Review by williamFalcon — APPROVED** [review]

**Pull request merged**